docs/guide/02-grammar-and-types.md

```diff
@@ -1,53 +1,152 @@
-# Context Free Grammar For Swahili
+# Grammar and Data Types
 
-## Variable Declaration BNF
-Wacha {var_name} =  {value}
+## Basics
+- Swahili borrows its syntax from javascript, python and ruby. Like Javascript, Swahili is case sensitive. as such the declarations
+``` wacha x = 5
+    wacha X = 3
+ ```
+ are entirely different variables. 
+ - Lines can optionally end with a semi-colon like in javascript. 
+ 
+## Comments
+ 
+## Declarations
+- To declare a variable, we use the keyword ```wacha```
+``` wacha x = 5
+    wacha X = 3
+ ```
+
+## Variables
+- Variables are used as symbolic names for values within the program. The names of variables, called **identifiers** must begin with either an underscore (```_``` ), or an alphabetic character in either upper or lower case. Subsequent characters can be alphanumeric or an underscore. Hence:
+```
+wacha _jina = "Kiano" //valid
+wacha Jina = "Kiano" //valid
+wacha jina = "Kiano" //valid
+wacha 0_jina = "Kiano" //invalid
+```
+- Swahili is case sensitive, hence, ```Jina``` and ```jina ``` are different variable. 

```

Comment:
different `variables`.

Also, for inline code elements, simply wrap using a single `

Reply:
Will work on that.